superx/information_extractor.py

```diff
@@ -139,15 +139,20 @@ def extract_information(self, items):
 
             unit_of_measure = self.convert_unit_name(item.find('UnitQty').text)
             # if item already in db then continue to next item
-            if Product.query.get(item_code):
+            if bool(Product.query.filter_by(id=item_code).first()):
+                # adding current_branch_price to the db and commit
+                current_branch_price = BranchPrice(item_code=item_code, price=price, update_date=update_date)
+                db.session.add(current_branch_price)
+                db.session.commit()
                 continue
+
```

Comment:
You only execute insert into BranchPrice if the product is already in the database. If it isn't you add the product to the database here, but don't add the Branch Price.
You probably want to check if the product exists. If not, add it. And then in any case add the branchPrice.